fsmonitor.c

```diff
@@ -149,15 +149,16 @@ void write_fsmonitor_extension(struct strbuf *sb, struct index_state *istate)
 /*
  * Call the query-fsmonitor hook passing the last update token of the saved results.
  */
-static int query_fsmonitor(int version, const char *last_update, struct strbuf *query_result)
+static int query_fsmonitor(int version, struct repository *r,
```

Comment:
I'm wondering if we should just take an `istate` here instead of having the caller pass both `istat->repo` and `istate->fsmonitor_last_update` ??

Reply:
Good idea!